File: School/TruDan.cpp
#include <iostream>
#include <stdio.h>
#include <fstream>
using namespace std;

const int N=100009;
int a[N], s[N], n, m;
//---------------------------------------------------------
void doc(){
	fstream f;
	int x;
	f.open("TruDan.inp", ios::in);
	f>> n; m=0;
	while(!f.eof()){
		f>> x;
		s[++m]=x;
	}
	f.close();
}
//---------------------------------------------------------
void xuly(){
	a[0]=0;
	for(int i=1; i<=n; i++){
		a[i]=0;
		for(int j=1; j<=m; j++)
			if (i-s[j]<0) break;
			else if (a[i-s[j]]==0){
					a[i]=1; break;
				}
	}
}
//---------------------------------------------------------
int main(){
	freopen("TruDan.out", "w", stdout);
	doc();
	xuly();
	cout<< a[n];
}

File: School/TruDan.pas
Const fi='TruDan.inp';
      fo='TruDan.out';
Var a,s:array[0..1000000] of integer;
    n,m:integer;
//---------------------------------------------------------
Procedure docfile;
    Var f:text;
    Begin
        assign(f,fi);
        reset(f);
        readln(f,n);
        m:=0;
        while not seekeof(f) do
            Begin
                inc(m); read(f,s[m]);
            End;
        close(f);
    End;
//---------------------------------------------------------
Procedure xuat;
    Var f:text;
    Begin
        assign(f,fo);
        rewrite(f);
        write(f,a[n]); write(a[n]);
        close(f);
    End;
//---------------------------------------------------------
Procedure xuly;
    Var i,j:integer;
    Begin
        fillchar(a,sizeof(a),0);
        for i:=1 to n do
            if a[i]=0 then
                for j:=1 to m do
                    if (i-s[j]>=0) and (a[i-s[j]]=0) then
                        Begin
                            a[i]:=1;
                            break;
                        End;
    End;
//---------------------------------------------------------
BEGIN
    docfile;
    xuly;
    xuat;
END.
